lib/actions/preload.coffee

```diff
@@ -0,0 +1,206 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+###
+
+_ = require('lodash')
+form = require('resin-cli-form')
+preload = require('resin-preload')
+Promise = require('bluebird')
+resin = require('resin-sdk-preconfigured')
+
+dockerUtils = require('../utils/docker')
+
+LATEST = 'latest'
+
+buildExpandOptions =
+	build:
+		$select: ['id', 'commit_hash', 'push_timestamp']
+		$orderby: 'push_timestamp desc'
+		$filter:
+			status: 'success'
+
+getApplicationsWithSuccessfulBuilds = (resin, deviceType) ->
+	resin.pine.get
+		resource: 'my_application'
+		options:
+			filter:
+				device_type: deviceType
+				build:
+					$any:
+						$alias: 'b'
+						$expr:
+							b:
+								status: 'success'
+			expand:
+				build:
+					$select: [ 'id', 'commit_hash', 'push_timestamp', 'status' ]
+					$orderby: 'push_timestamp desc'
+					# the filter is commented because it causes an api error
+					#$filter:
+					#	status: 'success'
+			select: [ 'id', 'app_name', 'device_type' ]
+			orderby: 'app_name asc'
+	# manual filtering
+	.then (applications) ->
+		applications.forEach (application) ->
+			application.build = application.build.filter (build) ->
+				build.status == 'success'
+		applications
+
+selectApplication = (resin, form, deviceType) ->
+	getApplicationsWithSuccessfulBuilds(resin, deviceType)
+	.then (applications) ->
+		if applications.length == 0
+			expectedError("You have no apps for a '#{deviceType}' device type.")
+		form.ask
+			message: 'Select an application'
+			type: 'list'
+			choices: applications.map (app) ->
+				name: app.app_name
+				value: app
+
+selectApplicationCommit = (resin, form, builds) ->
+	if builds.length == 0
+		expectedError('This application has no successful builds.')
+	DEFAULT_CHOICE = {'name': LATEST, 'value': LATEST}
+	choices = [ DEFAULT_CHOICE ].concat builds.map (build) ->
+		name: "#{build.push_timestamp} - #{build.commit_hash}"
+		value: build.commit_hash
+	return form.ask
+		message: 'Select a build'
+		type: 'list'
+		default: LATEST
+		choices: choices
+
+disableAutomaticUpdates = (form, resin, application, commit) ->
+	if commit == LATEST or not application.should_track_latest_release
+		return Promise.resolve()
+	message = '''
+
+		This application is set to automatically update all devices to the latest
+		available version.
+		You need to disable this if you want to preload a specific build.
+		Do you want to disable automatic updates for this application?
+	'''
+	form.ask
+		message: message,
+		type: 'confirm'
+	.then (update) ->
+		if not update
+			return
+		resin.pine.patch
+			resource: 'application'
+			id: application.id
+			body:
+				should_track_latest_release: false
+
+expectedError = (message) ->
+	console.error(message)
+	process.exit(1)
```

Comment:
Nice! This is a good solution to some of the annoying noise in sentry (as well as avoiding extra console noise, which I assume is what you're going for).

It'd be good if it didn't live here though - this should probably be in utils somewhere really.